modules/tinymce/src/plugins/help/main/ts/ui/PluginsTab.ts

```diff
@@ -55,14 +50,15 @@ const tab = (editor: Editor): Dialog.TabSpec => {
   const makeLink = (p: {name: string; url: string}): string =>
     `<a href="${p.url}" target="_blank" rel="noopener">${p.name}</a>`;
 
-  const maybeUrlize = (editor: Editor, key: string) => Arr.find(PluginUrls.urls, function (x: PluginUrlType) {
+  const maybeUrlize = (editor: Editor, key: string) => Arr.find(PluginUrls.urls, function (x: PluginUrls.PluginUrlType) {
     return x.key === key;
   }).fold(function () {
     const getMetadata = editor.plugins[key].getMetadata;
     return typeof getMetadata === 'function' ? makeLink(getMetadata()) : key;
   }, function (x) {
     const urlSlug = x.slug || x.key;
-    return makeLink({ name: x.name, url: 'https://www.tiny.cloud/docs/plugins/' + urlSlug });
+    const type = x.type || 'opensource';
```

Comment:
We lose a bit of type safety here instead of just making `type` compulsory and setting all open source plugins to `'opensource'`.

For example if we changed the interface definition to say, `readonly type?: 'open-source' | 'premium';` - this would still default to `'opensource'`

Reply:
That adds a lot of bloat, so I'm very much against that. That is actually what was here originally and I worked with Tyler to change it. I'm fine with this, as it's very low risk.

Reply:
Might it be better to remove `'opensource'` from the type if we never use it, which make it clearer that it's either `premium` or defined elsewhere?

Alternatively

`isPremium?: boolean`

And define the strings here, which would be even slimmer and mean changes occur in one place.

Reply:
`isPremium: true`
`type: 'premium'`

The current variation is actually the same size as that suggestion (it's slightly smaller once minified though). We could get the current version smaller than the new suggestion by using a constant for `premium` though. It also allows for other types in the future if needed.

If we're worried about types, then I'd prefer to solve it a different way (remap and return a `PluginUrlType` with the defaults), however it really feels like that's overkill here.

Reply:
Actually doing the map in `PluginUrls` would probably be a good idea, as then the defaults are all in the one location. I'll make that change, unless there's any objections?

Reply:
👍 I think having it in one place will be good

Reply:
Done. How does that look?